File: src/main/java/io/scriptor/result/StringResult.java
package io.scriptor.result;

import java.io.ByteArrayInputStream;
import java.io.InputStream;

public class StringResult extends ResultBase<String> {

    public StringResult(final int statusCode) {
        super(statusCode);
    }

    public StringResult(final int statusCode, final String statusText) {
        super(statusCode, statusText);
    }

    public StringResult(final int statusCode, final String statusText, final String value) {
        super(statusCode, statusText, value);
    }

    @Override
    public int getSize() {
        return getValue().getBytes().length;
    }

    @Override
    public InputStream getStream() {
        return new ByteArrayInputStream(getValue().getBytes());
    }
}

File: server/src/main/kotlin/dev/scriptor/server/result/StringResult.kt
package dev.scriptor.server.result

import dev.scriptor.server.ParameterList
import dev.scriptor.server.RangeReadableByteChannel
import java.nio.channels.Channels
import java.nio.channels.ReadableByteChannel

class StringResult : Result {

    private companion object {

        fun channel(value: String): RangeReadableByteChannel {
            val count: Long
            val channel: ReadableByteChannel?
            val array = value.encodeToByteArray()

            channel = Channels.newChannel(array.inputStream())
            count = array.size.toLong()

            return RangeReadableByteChannel(channel, 0L until count)
        }
    }

    constructor(
        statusCode: Int = 200,
        statusText: String = "OK",
        contentType: String = "text/plain",
        headers: ParameterList = ParameterList(),
        value: String,
    ) : super(
        statusCode,
        statusText,
        contentType,
        headers,
        channel(value),
    )
}
